x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/apikey/GetApiKeysRestIT.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.apikey;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.XContentTestUtils;
+import org.elasticsearch.transport.TcpTransport;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.core.security.action.apikey.ApiKey;
+import org.elasticsearch.xpack.core.security.action.apikey.GetApiKeyResponse;
+import org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken;
+import org.elasticsearch.xpack.security.SecurityOnTrialLicenseRestTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.emptyArray;
+import static org.hamcrest.Matchers.equalTo;
+
+public class GetApiKeysRestIT extends SecurityOnTrialLicenseRestTestCase {
+    private static final SecureString END_USER_PASSWORD = new SecureString("end-user-password".toCharArray());
+    private static final String MANAGE_OWN_API_KEY_USER = "manage_own_api_key_user";
+    private static final String MANAGE_SECURITY_USER = "manage_security_user";
+
+    @Before
+    public void createUsers() throws IOException {
+        createUser(MANAGE_OWN_API_KEY_USER, END_USER_PASSWORD, List.of("manage_own_api_key_role"));
+        createRole("manage_own_api_key_role", Set.of("manage_own_api_key"));
+        createUser(MANAGE_SECURITY_USER, END_USER_PASSWORD, List.of("manage_security_role"));
+        createRole("manage_security_role", Set.of("manage_security"));
+    }
+
+    public void testGetApiKeysWithActiveOnlyFlag() throws Exception {
+        final String apiKeyId0 = createApiKey(MANAGE_SECURITY_USER, "key-0");
+        final String apiKeyId1 = createApiKey(MANAGE_SECURITY_USER, "key-1");
+        // Set short enough expiration for the API key to be expired by the time we query for it
+        final String apiKeyId2 = createApiKey(MANAGE_SECURITY_USER, "key-2", TimeValue.timeValueNanos(1));
```

Comment:
It does feel like a bug, good catch! 

I haven't made up my mind yet about whether to fix it though. I can see some contrived scenario where customer is using `0`--or maybe even negative values if they're doing some date math--in e.g., some automated test pipeline and changing this breaks things for them. This feels very far-fetched though. 

On the other hand, the advantage of fixing this is also quite small. 

The only "real" scenario that comes to mind where someone might be using `0` or `-1` is if they're misinterpreting the API: `expiration=0` or `expiration=-1` could be interpreted as "never expire" -- if someone doesn't carefully read the docs, they might just assume that's what it means. But that's actually more of an argument for fixing the bug... 

TL;DR: I'm leaning (slightly) towards treating this as a bug, without BWC implications, and just fixing it, but I want to sit on this for a bit and see if I come up with a reason not to. What would be your preference?

For this PR, I will leave the `1ns` as is.

Reply:
My preference is also treating it as a bug since I cannot think of any useful thing it can do. Definitely not critical though because the ultimate behaviour is correct, i.e. the API key cannot be used. We may also want to get the team's consensus on this.